File: dk05.cpp
/*Hailun*/
#include<iostream>
#include<math.h>
using namespace std;
bool Check(long long x)
{
    long long tmp;
    tmp = sqrt(x);
    if(tmp * tmp == x)
        return true;
    return false;
}

int main()
{
    long long n;
    cin >> n;
    if(Check(n) == true)    cout << "YES" ;
    else    cout << "NO" ;
    return 0;
}

File: dk05.java
package luyencode;

import java.util.Scanner;

public class dk05 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long a = sc.nextLong();
        if (KTscp(a))
            System.out.println("YES");
        else
            System.out.println("NO");
        sc.close();
    }

    public static boolean KTscp(long a) {
        long b = (int) Math.sqrt(a);
        if (b * b == a)
            return true;
        return false;
    }
}
